pkg/server/handler/auth.go

```diff
@@ -580,8 +583,19 @@ func (h *PasswordHandler) Handle(payload *router.Payload, response *router.Respo
 		panic(err)
 	}
 
-	response.Result = AuthResponse{
-		UserID:      info.ID,
-		AccessToken: token.AccessToken,
+	user := payload.User
```

Comment:
Then I would return a ResourceNotFound error instead of panic, and instead of changing the behaviour of the user preprocessor as you suggested. This has the same behaviour with `InjectAuth`.